pipelines/ci-build.yml

```diff
@@ -1,226 +1,291 @@
+                              "parameterValue": "http://www.microsoft.com"
+                          },
+                          {
+                              "parameterName": "FileDigest",
+                              "parameterValue": "/fd \"SHA256\""
+                          },
+                          {
+                              "parameterName": "PageHash",
+                              "parameterValue": "/NPH"
+                          },
+                          {
+                              "parameterName": "TimeStamp",
+                              "parameterValue": "/tr \"http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer\" /td sha256"
+                          }
+                      ],
+                      "toolName": "sign",
+                      "toolVersion": "1.0"
+                  },
+                  {
+                      "keyCode": "CP-230012",
+                      "operationSetCode": "SigntoolVerify",
+                      "parameters": [],
+                      "toolName": "sign",
+                      "toolVersion": "1.0"
+                  }
+              ]
+            SessionTimeout: 20
+        # arguments are not parsed in DotNetCoreCLI@2 task for `pack` command, that's why we have a custom pack command here
+        - powershell: |
+            dotnet pack $env:BUILD_SOURCESDIRECTORY/src/Microsoft.Graph/Microsoft.Graph.Beta.csproj /p:IncludeSymbols=true /p:SymbolPackageFormat=snupkg --no-build --output $env:BUILD_ARTIFACTSTAGINGDIRECTORY --configuration $env:BUILD_CONFIGURATION
+          env:
+            BUILD_CONFIGURATION: $(BuildConfiguration)
+          displayName: 'dotnet pack'
+        - task: EsrpCodeSigning@3
+          displayName: 'ESRP NuGet CodeSigning'
+          inputs:
+            ConnectedServiceName: 'microsoftgraph ESRP CodeSign DLL and NuGet (AKV)'
+            FolderPath: '$(Build.ArtifactStagingDirectory)'
+            Pattern: '*nupkg'
+            signConfigType: inlineSignParams
+            inlineOperation: |
+              [
+                  {
+                      "keyCode": "CP-401405",
+                      "operationSetCode": "NuGetSign",
+                      "parameters": [ ],
+                      "toolName": "sign",
+                      "toolVersion": "1.0"
+                  },
+                  {
+                      "keyCode": "CP-401405",
+                      "operationSetCode": "NuGetVerify",
+                      "parameters": [ ],
+                      "toolName": "sign",
+                      "toolVersion": "1.0"
+                  }
+              ]
+            SessionTimeout: 20
+        - task: CopyFiles@2
+          displayName: 'Copy release scripts to artifact staging directory'
+          inputs:
+            SourceFolder: '$(Build.SourcesDirectory)'
+            Contents: 'scripts\**'
+            TargetFolder: '$(Build.ArtifactStagingDirectory)'
 
-- task: PublishBuildArtifacts@1
```

Comment:
since there's no more publish artifact step, aren't you going to miss the artifact on the next stage?

Reply:
This shouldn't be an issue. The equivalent of `1ES.PublishPipelineArtifact@1` is specificed as `outputs` in line 43. 
See https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/features/outputs for reference.

Reply:
I didn't realize from the documentation this could be specified "ahead of time". Resolving.